session/resthandler.go

```diff
@@ -227,12 +217,13 @@ func DefaultClient(allowUntrusted bool, state *State) (*http.Client, error) {
 							Timeout:   30 * time.Second,
 							KeepAlive: 30 * time.Second,
 						}
-						conn, err := dialer.DialContext(ctx, network, net.JoinHostPort(ip, port))
+						var conn net.Conn
+						conn, err = dialer.DialContext(ctx, network, net.JoinHostPort(ip, port))
 						if err == nil {
 							return conn, nil
 						}
 					}
-					return nil, nil
+					return nil, err
```

Comment:
This was the cause of the panic? Nil conn?

Reply:
yupp